Assignment_2-data-emma-luk.js

```diff
@@ -0,0 +1,101 @@
+﻿// Emma Luk
+// Assignments 2 Using the revealing module pattern, build a Date utility module 
+
+var myDate = (function iife() {
+    var displaysModuleDate = null;
+
+    //private function: date instantiation
+    function _initDate() {
+        return new Date();
+    }
+
+    //check here that date is set: if not, initialize date to now
+    function setDate(date) {
+        if (date === undefined && null) {
+            displaysModuleDate = _initDate();
+        }
+        else if (typeof date === 'number' && !isNaN(date)) {
+            displaysModuleDate = new Date(date);
+        }
+        else if (typeof date === 'object') {
+            displaysModuleDate = date;
+        }
+        else {
+            console.log("sorry!! wrong type");
+        }
+    }
+
+    function getDate(objFormat) {
+        if (displaysModuleDate === null && undefined) {
+            displaysModuleDate = _initDate();
+        }
+        if (objFormat.format === "formatted") {
+            return displaysModuleDate.getMonth() + " " + displaysModuleDate.getDate() + ", " + displaysModuleDate.getFullYear();
+        }
+        else if (objFormat.format === "milliseconds" || objFormat === undefined) {
+            return displaysModuleDate.getDate();
+        }
+        else {
+            console.log("sorry!! unknown format");

```

Comment:
If the user does not pass an argument into this function, you want to default to returning the millisecond value for the date.